ethcore/src/engines/null_engine.rs

```diff
@@ -105,9 +106,9 @@ impl Engine for NullEngine {
 		}
 
 		fields.state.commit()?;
-		match tracing_enabled {
```

Comment:
Actually it would be better to avoid collecting traces at all if tracing is disabled, not only discard them afterwise.